data/ConsumerDecorator/tests/ConsumerDecorator-specs.js

```diff
@@ -0,0 +1,99 @@
+import '@testing-library/jest-dom';
+import {render} from '@testing-library/react';
+import compose from 'ramda/src/compose';
+
+import ConsumerDecorator from '../ConsumerDecorator';
+import {AppContext, ProviderDecorator} from '../../ProviderDecorator';
+
+let componentAProps = [];
+let componentBProps = [];
+
+const componentAState = {
+	componentAState: 'componentAState'
+};
+const componentBState = {
+	componentBState: 'componentBState'
+};
+
+const BaseComponentA = (props) => {
+	componentAProps = props;
+
+	return <AppContext.Provider value="defaultValue"><div /></AppContext.Provider>;
+};
+
+const ComponentADecorator = compose(
+	ProviderDecorator({
+		state: componentAState
+	}),
+	ConsumerDecorator({
+		handlers: {
+			defaultHandler: jest.fn()
+		},
+		// added for code coverage purposes. `mount` is not accessible through props
+		mount: () => {},
+		mapStateToProps: () => ({
+			defaultState: componentAState.componentAState
+		})
+	})
+);
+
+const ComponentA = ComponentADecorator(BaseComponentA);
+
+// ComponentB created for testing `mapStateToProps` defined as an object
+const BaseComponentB = (props) => {
+	componentBProps = props;
+
+	return <AppContext.Provider value="test"><div /></AppContext.Provider>;
+};
+
+const ComponentBDecorator = compose(
+	ProviderDecorator({
+		pauseOnBlur: true,
```

Comment:
Do we need `pauseOnBlur` for this test case?

Reply:
Yes, we do.
This line was added for code coverage purposes. This prop covers lines 37-38 andn 44-45 for ProviderDecorator.js
Removing this line wou;d reduce code coverage to 60%:
![image](https://user-images.githubusercontent.com/63335823/229079194-2b2134df-02a8-4f1f-9555-496bfe32b6ed.png)

Reply:
I'm afraid I cannot agree that pauseOnBlur is related to this test case. It can drop the code coverage of ProvideDecorator, but this test case is for testing ConsumerDecorator logic. And I think this TC is not designed for pauseOnBlur: true case(line 37-38).

I think to keep pauseOnBlur: true, we need to include test code like the below for example.
`expect(window.addEventListener).toBeCalledWith('blur', expect.any(Function));`

Reply:
I tried writing a TC for `pauseOnBlur` prop, but I can't find a way to access it.
So I think we can go ahead with your suggestion and remove it. Thank you!